src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -236,6 +236,26 @@ function validatePropTypes(element) {
   }
 }
 
+/**
+ * Given an element, freeze `type.defaultProps` (Only used in DEV).
+ *
+ * @param {ReactElement} element
+ */
+function freezeDefaultProps(element) {
+  var type = element.type;
+  if (typeof type !== 'function' || type.wasTaggedReactComponent) {
+    return;
+  }
+  Object.defineProperty(type, 'wasTaggedReactComponent', {value: true});
+  Object.defineProperty(type, 'defaultProps', {
+    writable: false,
+    configurable: false,
+  });
+  if (Object.freeze) {
```

Comment:
note: c508d90 also removes logic to reset `ReactDebugCurrentFrame.element` (since fiber/debug tracking has been decoupled from core implementation into renderers 3b5e3b5a23069dd03feaa6979bccb64de4d76d09)  ~ L255:

```diff
   if (Object.freeze) {
-    ReactDebugCurrentFrame.element = null;
     Object.freeze(type.defaultProps);
   }
```

i lost commit somehow in rebase, not too important but documenting for posterity